analyzers/src/SonarAnalyzer.Common/Extensions/AttributeDataExtensions.cs

```diff
@@ -46,6 +46,25 @@ public static bool TryGetAttributeValue<T>(this AttributeData attribute, string
             }
         }
 
+        /// <summary>
+        /// Returns the <see cref="AttributeUsageAttribute.Inherited"/> setting for the attribute associated with <paramref name="attribute"/>.
+        /// The returned value is in line with the runtime behavior of <see cref="System.Reflection.MemberInfo.GetCustomAttributes(bool)"/>.
+        /// </summary>
+        /// <param name="attribute">The <see cref="AttributeData"/> of an <see cref="Attribute"/>.</param>
+        /// <returns>
+        /// The <see cref="AttributeUsageAttribute.Inherited"/> value of the <see cref="AttributeUsageAttribute"/> applied to the <paramref name="attribute"/>.
+        /// Returns <see langword="true"/> if no <see cref="AttributeUsageAttribute"/> is applied to the <see cref="Attribute"/> associated with <paramref name="attribute"/>
+        /// as this is the default behavior of <see cref="System.Reflection.MemberInfo.GetCustomAttributes(bool)"/>.
+        /// </returns>
+        public static bool AttributeUsageInherited(this AttributeData attribute) =>
+            attribute.AttributeClass.GetAttributes()
+                .Where(x => x.AttributeClass is { Name: nameof(AttributeUsageAttribute), ContainingNamespace.Name: "System" })
+                .SelectMany(x => x.NamedArguments.Where(x => x.Key == nameof(AttributeUsageAttribute.Inherited)))
+                .Select(x => x.Value)
+                .Where(x => x is { Kind: TypedConstantKind.Primitive, Type.SpecialType: SpecialType.System_Boolean })
+                .Select(x => (bool?)x.Value)
+                .FirstOrDefault() ?? true; // Default value of Inherited is true
```

Comment:
Yes and no. There can be multiple `[AttributeUsage]` present, but it would be a compile error because "AllowMultiple" is false for `[AttributeUsage]`. But in the IDE we should expect compiler errors so it is safer to keep FirstOrDefault.

Reply:
Do we have UT for it?

Reply:
I added `HasAttributeUsageInherited_DuplicateAttributeUsage` in 8810dffd813e3c3469e407cb192033ccbd979b98